src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private int capacity;
+    private Object[] elements;
+
+    public ArrayList() {
+        capacity = DEFAULT_CAPACITY;
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int capacity) {
+        this.capacity = capacity;
+        elements = (T[]) new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= capacity) {
+            increaseCapacity();
+        }
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexForAdd(index);
+        if (size + 1 >= capacity) {
```

Comment:
The magic number is here `size + 1 >= capacity`. 

Introduce a constant with a proper name, or think about changing the condition.